packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -311,6 +346,10 @@ export class ChildLogger extends TelemetryLogger {
 export class MultiSinkLogger extends TelemetryLogger {
     protected loggers: ITelemetryBaseLogger[] = [];
 
+    // Multi-sink is a special case where messages are meant to be routed to multiple loggers, which may have different
+    // permissions. It should then be permissive by default, and let
+    supportsTags: true | undefined = true;
```

Comment:
MultiSinkLogger is a good example why I think handling of tags should not be based into these classes (as it may work with two loggers with different properties). Instead I'd go for 
1. There is an adapter ITelemetryBaseLogger implementation that can do proper stripping of data. It can be injected on as needed bases into the chain, without each class of the chain doing same thing.
   - adapter pattern allows usage of different adapters, and injection in the stack on as needed bases. It's easier to see intention / adjust logic when and how adapter should be used or not.
2. As to where it's injected - I'd say only where it matters. I believe the only place where it matters is loader to container runtime boundary, where either
   - container runtime adds this wrapper for the logger provided by loader unconditionally
   - or we do not use supportTags property, and instead rely on two properties on IContainerContext to communicate if logger has logger that supports tagging or not, and container runtime could wrap it or not depending on which property is set.

Reply:
If I am understanding you correctly, then I don't think the adapter pattern would work properly (though it would be more ideal). Since we have already committed to having supportsTags as a field on the base interface, we will always have to do this handshake between loggers where the first logger in a chain must check whether its recipient can support tags and then adjust the event accordingly; even the adapter logger implementation would need to specify supportsTags. Hypothetically if we went back in time and instead of adding this field as a property instead added some function sendTagged() that could allow loggers to be more agnostic about event objects themselves, that would be the right context for this approach - but it would be pretty complicated to do that as well as bohemia as already integrated the change to the base interface :(

CC @markfields

Reply:
It's not too late. And it's not that hard to change. Here are exact steps:
- Remove supportTags property form interface.
- Make host logger interface support tags (it can be as simple as noop + breaking.md note, though other options are available)
- loader provides that interface in IContainerContext.taggedLogger
- Container Runtime
    - uses IContainerContext.taggedLogger if it's provided
    - If not, takes IContainerContext.logger and wraps it with adapter that processes tagged properties and strips content on as needed bases.
- After a year, remove IContainerContext.logger property and handling of it.

Pros:
- There is no ambiguity: host has to support tagging, and it has to happen on picking up next version.
   - We can provide adapters to strip tags such that host who has no concerns convert tagged input into untagged, though I do not see much value (other than for staging), as it's just ignoring tags.
- There is no chance of some layer forgetting to set or propagate supportTags property and losing telemetry data.
- Interfaces and semantics are clearer.
Con:
- We force people to start supporting tags (i.e. change in behavior for current API).
   - I do not personally consider it's a big con, as even with continuing current flow, I'd love to see supportTags eventually removed from API (as well as our fallback mechanism)
- It's easy to not pay attention to breaking.md and not react.

Reply:
I don't have time to dig in on this comment now, I just skimmed it.  But I will say - I wish we _had_ made it a hard cut-over breaking change.  But now it's "too late" in that any further changes (e.g. https://github.com/microsoft/FluidFramework/pull/6444) won't get picked up for several months (TBD) since we somewhat froze those definitions interfaces.

Reply:
@vladsud, @ksbrar - Ok I see the point, this is a very nice pragmatic approach. I had been trying to think about transitioning in an "air-tight" way that would ensure we NEVER send tagged data to an old logger - regardless of the layer boundary - but in practice the only boundary that matters is Loader-Runtime, which can be mediated by this `IContainerContext` change. While discussing the `supportsTag` approach with Kabir I began to see what a confusing API design it is, so I'm glad to see a more straightforward alternative.

Looking at your list of steps, (1) is essentially optional and (2) is done for Bohemia (they drop tagged props).  So let's proceed with (3), including backporting to 0.39, and then (4) will functionally replace this PR.

When we're done, the main difference from the current approach will be that everywhere _except_ for that `IContainerContext`/runtime boundary, all loggers everywhere will assume everyone else supports tags and pass around tagged props freely. This seems like a fair compromise for a simpler approach.  Thanks for your help here!

Reply:
Ok but actually before doing (3) we need to _properly_ add tag support in Bohemia, otherwise existing tagged props like packageName will get dropped instead of being passed through.

Reply:
Vlad and I spoke this morning, here's the plan:

1. Work with Office Fluid team to properly support tags, specifically to pass-through `PackageData`. At this point though, no tagged data will come to their logger due to existing accommodation.
2. Patch the Definitions packages:
  a. `common-definitions`: Mark `supportsTags` as deprecated, or just remove it (if ok breaking semver rules).
  b. `container-definitions`: Add `taggedLogger`
3. Update Loader and Runtime
  a. Loader: pull in updated definitions, and duplicate `logger` to `taggedLogger` on `ContainerContext`.
  b. Runtime: Send tagged data as-is to `IContainerContext.taggedLogger` if present, otherwise continue handling tags and pass untagged data to `IContainerContext.logger`

Along the way, we'll need to revisit `BREAKING.md` to clarify all this.

FYI @ksbrar